### ISSUE TITLE
Update Fingerprint (Forest Client) work space Roles after AWS User Access Role Account Mapping is Complete

### ISSUE THREAD

**Issue by ArogeG** [bcgov/nr-forests-access-management#537]
Update AWS workspace to allow for segregated admin access across different environment. Request for access to be changed on Rocket Chat. 

**Role Mapping is as below:** 
Proposed Changes to User Access
Developer  Role --- No production access
Admin role --- includes access to Production

![Useraccess name.PNG](https://images.zenhubusercontent.com/62a1029ce9497c39c087ab8b/c908336a-7afb-49af-bf5d-4ca6ae4a967d)

**Comment by MCatherine1994** [comment]
Email sent to Pathfinder team

**Comment by MCatherine1994** [comment]
Currently Michelle has the BillingViewer role and rest of us Ian, Basil, Conrad, Pete, Michelle, Derek and me all have the admin access. I guess we change to:

BillingViewers:
Michelle,
Pete or Olga

Admins:
Conrad,
Basil,
Catherine

Developers (can access to all env other than prod):
Ian,
Derek

Readers:
Pete,
Olga

**Comment by ConradBoydElliottGustafson** [comment]
I wouldn’t trust myself from PROD if I’m not a full-time member of the team!

**Comment by basilv** [comment]
I somewhat agree with Conrad. Lets look at it a different way - what are the criteria _why_ someone should have prod admin access (read access to prod is also potentially harmful if it allows them to read secret values. A fundamental security principle is to grant the minimum access required for a role. Just because Conrad and myself are architects doesn't automatically mean we should have prod access. What are scenarios were prod admin access is required?
1. To create new prod secrets (or rotate prod secrets) - this is an ops activity, don't need an architect. This is a (infrequent) day-to-day activity that likely justifies an ops person to have prod admin access.
2. To make emergency changes for critical incidents. E.g. disable API gateway for cyber attack, or enable bastion host to do an emergency data fix. For this we should have an escalation procedure to gain prod access, or the ops person if they don't know can screenshare to be guided by an architect on what to do.

When might prod read access be required?
1. To troubleshoot issues. E.g. access to log files, in particular. For FAM, I think its reasonable to let most if not all developers have access to log files, although I don't think all developers should have access to prod secrets. (I'm not sure how the ASEA permissions are configured with respect to that). (For apps with sensitive data that might be shown in log files, developers often  don't get regular access. Which complicates troubleshooting.) I'd typically always give architects this read access.
2. To inspect configuration. E.g. AWS resource review. This is an ops role.

So to summarize the above, I'm saying that staff in the FAM ops role should have AWS admin and staff in the FAM architect/developer role should have the AWS developer role and the AWS reader role (confirm ideally no secret value access)

(I know we don't currently identify people as being in the 'ops' role. Scrum/DevOps has a viewpoint of the whole team supporting the app. This viewpoint can still be supported - any and all architects/developers can help when there's a critical or major incident, even though for security reasons admin access to prod has been limited to a subset (the ops role).

**Comment by MCatherine1994** [comment]
Thanks so much @ConradBoydElliottGustafson and @basilv !!

Updated:

BillingViewers:
Michelle
Pete (as Ops lead for now)

Admins:
Pete (as Ops lead to manage secrets)
Catherine (in the case some other teams need to change redirect urls, this is a temporary need, if Pete can support this then I don't need to have admin access)

Developers (can access to all env other than prod):
Ian
Derek
Basil
Conrad

Readers:
Basil (troubleshoot issues)
Conrad (onboarding teams to FAM and help debug onboarding issues, or Catherine can do this, Conrad can just have developer role)

I'll ask the pathfinder team to check how long it takes them to grant admin access in the case of emergency.

**Comment by ConradBoydElliottGustafson** [comment]
Good writeup. We should ask the cloud pathfinder team if the developer role can read secrets. The risk is that they create a jump box into the database and go to town.

**Comment by MCatherine1994** [comment]
Request sent to pathfinder team:

BillingViewers:
Michelle Douville (stay as the same)

Admins:
Pete Tollestrup (stay as the same)
Catherine Meng (stay as the same)

Developers:
Ian Liu
Derek Robert
Basil Vandegriend
Conrad Gustafson

Readers:
Basil Vandegriend
Conrad Gustafson
